app/javascript/gobierto_data/lib/commons.js

```diff
@@ -168,3 +170,29 @@ export const sqlKeywords = [
   }
 ]
 
+export const convertVizToImgMixin = {
+  methods: {
+    data() {
+      return {
+        imageApi: null
+      }
+    },
+    convertVizToImg(opts) {
+      let node = document.querySelector('.gobierto-data-visualization--aspect-ratio-16-9');
+      const perspectiveChart = document.querySelector("perspective-viewer").shadowRoot
+      const perspectiveSidePanel = perspectiveChart.getElementById("side_panel")
+      const perspectiveTopPanel = perspectiveChart.getElementById("top_panel")
+      perspectiveSidePanel.style.display = "none"
+      perspectiveTopPanel.style.display = "none"
+      htmlToImage.toPng(node)
+        .then(function (dataUrl) {
+          this.saveLoader = true
+          this.imageApi = dataUrl
+          this.onSaveEventHandler(opts)
```

Comment:
And this? what's this about `onSaveEventHandler`? 😲 Bad practices.

Move the complete callback function to the actual component who uses it

Reply:
done!